cas/service.c

```diff
@@ -1913,30 +1913,30 @@ static int cas_device_check(const struct cas_fom   *fom,
 static int cas_id_check(const struct m0_cas_id *cid, const struct cas_fom *fom)
 {
 	const struct m0_dix_layout *layout;
-	int                         rc = 0;
+	int                         rc;
 	struct cas_service         *svc;
-	uint32_t                    device_id = 0;
+	uint32_t                    device_id;
 
+	rc = 0;
 	if (!m0_fid_is_valid(&cid->ci_fid) ||
 	    !M0_IN(m0_fid_type_getfid(&cid->ci_fid), (&m0_cas_index_fid_type,
 						      &m0_cctg_fid_type)))
-		rc = M0_ERR(-EPROTO);
+		return M0_ERR(-EPROTO);
 
 	if (rc == 0 && cas_fid_is_cctg(&cid->ci_fid)) {
 		layout = &cid->ci_layout;
 		if (layout->dl_type != DIX_LTYPE_DESCR)
-			rc = M0_ERR(-EPROTO);
+			return M0_ERR(-EPROTO);
 		if (fom != NULL) {
 			svc = M0_AMB(svc, fom->cf_fom.fo_service, c_service);
 			device_id = m0_dix_fid_cctg_device_id(&cid->ci_fid);
 			if (svc->c_sdev_id != INVALID_CAS_SDEV_ID &&
 			    svc->c_sdev_id != device_id &&
 			    cas_type(&fom->cf_fom) != CT_META) {
-				M0_ASSERT_INFO(svc->c_sdev_id == device_id,
-				       	       "Incorrect device ID (%d) found "
-					       "in component catalogue FID. "
-					       "Valid device ID should be %d", 
-					       device_id, svc->c_sdev_id);
+				M0_LOG(M0_INFO, "Incorrect device ID (%d) "
```

Comment:
Please use M0_ERR_INFO() here to use the shorter form (error log + assignment), example:
rc = M0_ERR_INFO(-EPROTO, "Incorrect device ID ...");